kedro/extras/datasets/spark/spark_dataset.py

```diff
@@ -309,7 +309,13 @@ def __init__(  # pylint: disable=too-many-arguments
             path = PurePosixPath(filepath)
 
             if filepath.startswith("/dbfs"):
-                dbutils = _get_dbutils(self._get_spark())
+                dbutils = None
+                try:
+                    dbutils = _get_dbutils(self._get_spark())
+                except AttributeError:
+                    # Databricks is known to raise AttributeError when called
+                    # on an unsupported environment
+                    pass
```

Comment:
I am not sure I understand the root cause entirely. Is this a bug from Databricks `pyspark.dbutils` module or is it because we check the filepath too eagerly in kedro?

The `_get_dbutils` function is suppose to try getting the dbutils aggressively, if not it just return `None`. This solution is adding yet another try-catch layer outside is a bit hacky but maybe necessary in this case? I want to make sure I understand the problem before I come to the conclusion.

Is it better to have this try-except block inside the `_get_dbutils` func if necessary?

Reply:
I believe it's a bug in DataBricks code. It assumes that `IPython.get_ipython()` returns an object. When it happens to return None, we get an `AttributeError`.

```python
│ /databricks/spark/python/pyspark/dbutils.py:50 in get_dbutils                │
│                                                                              │
│    47 │   │   │   return SparkServiceClientDBUtils(spark.sparkContext)       │
│    48 │   │   else:                                                          │
│    49 │   │   │   import IPython                                             │
│ ❱  50 │   │   │   return IPython.get_ipython().user_ns["dbutils"]            │
│    51                                                                        │
│    52                                                                        │
│    53 class SparkServiceClientDBUtils(object):                               │
```

I think having the try-except block inside `_get_dbutils` is a better solution indeed. Thanks for pointing that out!

Reply:
Is this `pyspark.dbutils` module only available on Databricks runtime? If so I think that's why it assumes you have `IPython`. Also, you mentioned you are running on Databricks but not in a managed way, I am not aware that there is an on-premise option, what kind of environment are you running on?

Reply:
I was trying to run on a normal DataBricks cluster, just with MLFlow instead of a notebook. I managed to run pipelines via a notebook too but it would have been better to do it through command line. So, when I run `mlflow run`, MLFlow packages my project into a zip file, sends to a new DataBricks cluster, and runs it on there. Apparently, because it's not on a notebook, there's no IPython.

If you think that this use case is worth it to support, I can make the change that you proposed.

Reply:
@mle-els I prefer moving the try-except block to `_get_dbutils`. For `IPython` I am unsure, even running with `.py` file it will have `IPython` normally, but Databricks doesn't document.

Reply:
I used MLFlow 1.28.0 and a DataBricks cluster with runtime 9.1. You could try it out yourself, if that's possible?

Reply:
@mle-els  I won't be able to test it myself since I don't have the environment configured. My guess will be you have a relatively old Databricks runtime.

We test it recently with `dbx`, which package up a project and runs as a Databricks Job, and IPython would be available in that case.

This suggest Databricks runtime >11 always run on IPython, although it mentioned notebook only but as we tested a couple months ago, it's the same with .py file.
https://docs.databricks.com/notebooks/ipython-kernel.html#how-to-use-the-ipython-kernel-with-databricks

Reply:
This week I'm swamped, unfortunately :( Please feel free to close it.

Reply:
@mle-els No worries! Feel free to re-open the PR in the `kedro-plugins` repository when you are free to work on it again. :)